### ISSUE TITLE
App not working

### ISSUE THREAD

**Issue by aayushbhatia06** [steventheworker/Dock-Expos--home#5]
Mac OS Ventura
M2 Macbook Air 

App doesn't work, gave it permissions in Settings < Privacy & Security < Accessibility 
Settings < Privacy & Security < Input Monitoring 

Just dosent do anything

**Comment by MotherOfClamperl** [comment]
1. Remove permissions from Accessibility & Input Monitoring
2. Remove Dock Exposé from filesystem by trashing & emptying from trash (so that it doesn't exist in spotlight)
3. Restart
4. Redownload Dock Exposé, place in Applications (give permissions BEFORE running)

**Comment by steventheworker** [comment]
1. Remove permissions from Accessibility & Input Monitoring
2. Remove Dock Exposé from filesystem by trashing & emptying from trash (so that it doesn't exist in spotlight)
3. Restart
4. Redownload Dock Exposé, place in Applications (give permissions BEFORE running)

**Comment by aayushbhatia06** [comment]
Did that, still nothing. Worth mentioning that I had to manually add the app to both permissions by clicking the plus icon.
My Macbook did NOT come with Ventura

**Comment by MotherOfClamperl** [comment]
Are you on v2.9 or v2.91? Did it use to work and just stop after restarting?

**Comment by steventheworker** [comment]
Are you on v2.9 or v2.91? Did it use to work and just stop after restarting?

**Comment by aayushbhatia06** [comment]
Upon further investigation, hovering over closed apps on the dock (i.e. with no open windows) the computer makes a tap sound (I dont know how to describe it, kind of like "tak"). Doesn't happen on apps on dock with open Windows. 

Also, fwiw, Double tap on app icon does work and does open expose.

I tried both 2.9 and 2.91, no dice on either

I do use "Rectangle", "Expanso" and "Pure Paste". Did try closing all of them though still no dice.

**Comment by steventheworker** [comment]
That sound is from sending the escape key to close Exposé because it *should* be open. Escapes are only triggered when your mouse leaves the dock directly after hovering an icon with open Windows.

2.9 is gonna be easier to install because you don't need to bypass gatekeeper (2.91 has not been notarized).

I looked into those apps and those aren't causing it. Do you use any apps to reverse scroll direction apps?
Are you using "Natural Scrolling" in Settings -> Trackpad/Mouse -> Scroll & Zoom?

By double tap on app icon do you mean right click -> Show All Windows?

**Comment by aayushbhatia06** [comment]
By double tap I meant that if you double tap (NOT click) any icon on the dock it open expose menu. Built in functionality in Mac OS

On the other hand, the app "UnnaturalScrollWheels" was causing this app to not work. Closing that app fixes this one (didnt notice before because I had the menu bar icon off). Is there no way to keep both apps working? Because I use that one for my mouse

**Comment by steventheworker** [comment]
Just finished patching v2.91 to work with UnnaturalScrollWheels. If you were using 2.9 you might want to readd 
permissions (otherwise ignore this)

**Comment by aayushbhatia06** [comment]
So v2.9 works fine on my device, however 2.91 simply doesn't work. Tried all the troubleshoothing steps from above

**Comment by steventheworker** [comment]
I'm thinking this is because v2.91 is unnotarized. This means if you have the signed app working, the permissions will not apply to the unsigned version and you need to remove all permissions (Input Monitoring, Accessibility) manually or run ```tccutil reset All com.steventheworker.Dock-Expose```.  v2.91 might also work without adding the permissions beforehand, since I also recently made it popup and ask for both permissions.

As a last resort: the two scroll reversers I know work with v2.9 is ["Mos"](https://mos.caldis.me/) (I personally use this), and I had this same issue reported (and fixed it) for ["Scroll Reverser"](https://pilotmoon.com/scrollreverser/).

**Comment by aayushbhatia06** [comment]
The terminal command worked flawlessly! And the app is now also working with UnnaturalScrollWheels. Thank you so so much!

**Comment by steventheworker** [comment]
Great to hear! And you shouldn't have to do this moving forward, until new versions are notated again.

**Comment by SylvainMartel** [comment]
I had the same problem with 2.92.  Solved by running the command shown above
